### PR TITLE
Adding dictionary appending functions

### PR DESCRIPTION
**Pull request by danielaRiesgo**
## Summary ##

Adding a function to append dictionary contents (came up in training).
Also associating Extension folders to the file systems folders (that's why the changes in the project file).

### CODE REVIEW

**Comment by woloxci** [comment]
:+1: There are no linter violations.

File: Core/Extensions/Foundation/Dictionary.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        result.appending(contentsOf: dictionary)
+        return result
+    }
+    
+    /**
+         Adds to self the contents of the given `dictionary`.
+         If both dictionaries have the same key with different value,
+         the result will have that key with the value found in `dictionary`.
+     */
+    mutating func appending(contentsOf dictionary: Dictionary<Key, Value>) {
```
Comment (pablog) [inline Core/Extensions/Foundation/Dictionary.swift]: Why `append(contentsOf dictionary: Dictionary<Key, Value>)` returns a new dictionary? and this one appends in the current one?

Comment (danielaRiesgo) [inline Core/Extensions/Foundation/Dictionary.swift]: It should be the other way around, right?

File: Core/Extensions/Foundation/Dictionary.swift [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+extension Dictionary {
+    
+    /**
+        Returns a new dictionary with the contents of self and the given `dictionary`.
+        If both dictionaries have the same key with different value, the result
+        will have that key with the value found in `dictionary`.
+    */
+    func appending(contentsOf dictionary: Dictionary<Key, Value>) -> Dictionary<Key, Value> {
+        var result = self
+        result.append(contentsOf: dictionary)
```
Comment (damfinkel) [inline Core/Extensions/Foundation/Dictionary.swift]: Does swift know that the Dictionary that receives the message must be Dictionary<Key, Value> ?

File: Core/Extensions/Foundation/Dictionary.swift [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+extension Dictionary {
+    
+    /**
+        Returns a new dictionary with the contents of self and the given `dictionary`.
+        If both dictionaries have the same key with different value, the result
+        will have that key with the value found in `dictionary`.
+    */
+    func appending(contentsOf dictionary: Dictionary<Key, Value>) -> Dictionary<Key, Value> {
+        var result = self
```
Comment (damfinkel) [inline Core/Extensions/Foundation/Dictionary.swift]: why is this necessary?

Comment (danielaRiesgo) [inline Core/Extensions/Foundation/Dictionary.swift]: To create a copy of the value type, since the append function is mutating.

File: Core/Extensions/Foundation/Dictionary.swift (hunk 2 above)
Comment (danielaRiesgo) [inline Core/Extensions/Foundation/Dictionary.swift]: Yes, since here `Key` and `Value` are typealias that exists in the `Dictionary` type (see that the function is not generic) and we can reach them since we are "inside" the `Dictionary` type in the extension.
